packages/mongoose/src/generateModel.ts

```diff
@@ -125,7 +125,8 @@ export const generateSchema = <T>(
 		const schemaDecoration = classReflection.decorators.find(d => d[DecoratorId] === 'mongoose.schema');
 		const schemaOptions = schemaDecoration?.options;
 		const schema: Schema<U> =
-			(forceCreateSchema || schemaDecoration) && new Schema(definition, { ...(options ?? schemaOptions) });
+			(forceCreateSchema || schemaDecoration) &&
+				new Schema(definition, { ...({ ...DEFAULT_SCHEMA_OPTIONS, ...options } ?? schemaOptions) });
```

Comment:
I suggest performing a deep merge, while we are here ([example](https://github.com/HPInc/davinci/blob/master/packages/core/src/entity/EntityDefinition.ts#L15))